JavadocStyleGuide.md

```diff
@@ -0,0 +1,242 @@
+| 3 | `@implSpec` | Adds a “Implementation Requirements” heading | This is where the default implementation (or an overrideable implementation in a class) is specified.                                                                                                               |
+| 4 | `@implNote` | Adds a “Implementation Note” heading         | This section contains informative notes about the implementation, such as advice to implementors, or performance characteristics that are specific to the implementation in this class of this version of the JDK.  |
+
+When a block tag doesn't fit on a single line, continuation lines are indented four (or more) spaces from the position of the `@`.
+
+## 1.2 The summary fragment
+
+Each Javadoc block begins with a brief summary fragment - this is the first sentence up until the character `.` is encountered.
+
+This fragment is very important: it is the only part of the text that appears in certain contexts such as class and method indexes:
+
+```java
+/**
+ * Returns an Image object that can then be painted on the screen.
+ */
+```
+
+## 1.3 Code blocks and snippets
+
+`<pre>` is the default HTML tag for preformatted text.
+All code blocks and snippets should be wrapped in `<pre>...</pre>` at minimum.
+
+Code blocks are formatted using [`highlight.js`](https://highlightjs.org/) using the `<code>` tag.
+All code blocks and snippets should be wrapped in `<code class="language-*****"></code>` to be automatically highlighted.
+Make sure the correct language is added to the class name: `language-kotlin`, `language-groovy`, or `language-java`.
+
+A formatted and highlighted code snippet looks should look as follows:
+
+```java
+/**
+ * <pre class='groovy autoTested'><code class="language-kotlin">
+ * project.ext.prop1 = "foo"
+ * task doStuff {
+ *     ext.prop2 = "bar"
+ * }
+ * subprojects { ext.${prop3} = false }
+ * </code></pre>
+ */
+```
+
+In order to label the coding language used in the snippet, you can use additional HTML, `<div class="code-block"><span class="label">Groovy</span>...</div>`:
+
+```java
+/**
+ * <div class="code-block">
+ * <span class="label">Groovy</span>
+ * <pre class='groovy autoTested'><code class="language-groovy">
+ * defaultTasks('some-task')    // Delegates to Project.defaultTasks()
+ * reportsDir = file('reports') // Delegates to Project.file() and the Java Plugin
+ * </code></pre>
+ * </div>
+ */
+```
+
+## 1.4 Where Javadoc is used
+
+At the minimum, Javadoc is present for every public class, and every public or protected member of such a class, with a few exceptions such as overrides and self-explanatory members:
+
+```java
+public Image getImage(URL url, String name) {}
+```
+
+## 1.5 A note on IDEs
```

Comment:
Maybe it's worth adding another sub-section on how to best configure the IDE to make it easier to follow the style guide.

Things like https://github.com/gradle/gradle/pull/28712/files#r1619850293 should go in there, so that everyone can be pointed at those settings. That is, if they cannot be shared via a config file checked into git.